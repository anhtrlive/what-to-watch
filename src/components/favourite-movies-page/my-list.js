import PropTypes from "prop-types";
import React from 'react';
import {FilmsList} from '../moviecard-films-list/films-list';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import {Operation} from '../../reducer';
import FooterLogo from "../footer/footer-logo";
import HeaderLogo from "../header/header-logo";
import HeaderAvatar from "../header/header-avatar";

const MyList = (props) => {
  let {bookmarks, films, isAutorized} = props;

  if (films) {
    films = films.filter((movie) => {
      return bookmarks.some((el) => el === movie.name);
    });
  }
  if (!isAutorized) {
    return <Redirect to="/login" />;
  }
  return (
    <div className="user-page">
      <header className="page-header user-page__head">
        <HeaderLogo />
        <h1 className="page-title user-page__title">My list</h1>
        <HeaderAvatar />
      </header>
      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <FilmsList
          movies={films}
          genre={`my-list`}
        />
      </section>
      <FooterLogo />
    </div>
  );
};

MyList.propTypes = {
  bookmarks: PropTypes.array.isRequired,
  films: PropTypes.array.isRequired,
  isAutorized: PropTypes.bool.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  loadFavoriteMovies: () => {
    dispatch(Operation.loadFavoriteMovies());
  }
});

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    bookmarks: state.bookmarks,
    films: state.films,
    isAutorized: state.isAutorized,
  });
};


export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
