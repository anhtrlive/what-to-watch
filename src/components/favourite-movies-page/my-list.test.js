import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {films} from '../../mocks/films';
import {MyList} from './my-list';
import {BrowserRouter} from 'react-router-dom';

Enzyme.configure({adapter: new Adapter()});

const bookmarks = [films[0].name, films[1].name, films[2].name];
const isAutorized = true;

it(`button plus list`, () => {
  const tree = renderer.create(
      <BrowserRouter>
        <MyList
          bookmarks={bookmarks}
          isAutorized={isAutorized}
          films={films}
        />
      </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
