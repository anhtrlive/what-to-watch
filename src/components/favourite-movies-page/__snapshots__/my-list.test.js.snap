// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button plus list 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/avatar.jpg"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/the-grand-budapest-hotel/overview"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="img/the-grand-budapest-hotel.jpg"
              src="https://some-link"
              style={
                Object {
                  "objectFit": "fill",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            The Grand Budapest Hotel
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/the-grand-budapest-hotel/overview"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="img/the-grand-budapest-hotel.jpg"
              src="https://some-link"
              style={
                Object {
                  "objectFit": "fill",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            The Grand Budapest Hotel
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/the-grand-budapest-hotel/overview"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="img/the-grand-budapest-hotel.jpg"
              src="https://some-link"
              style={
                Object {
                  "objectFit": "fill",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            The Grand Budapest Hotel
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/the-grand-budapest-hotel/overview"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="img/the-grand-budapest-hotel.jpg"
              src="https://some-link"
              style={
                Object {
                  "objectFit": "fill",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            The Grand Budapest Hotel
          </h3>
        </a>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <a
          className="small-movie-card__link"
          href="/the-grand-budapest-hotel/overview"
          onClick={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="img/the-grand-budapest-hotel.jpg"
              src="https://some-link"
              style={
                Object {
                  "objectFit": "fill",
                }
              }
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            The Grand Budapest Hotel
          </h3>
        </a>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
